future-apps/payroll/contracts/Payroll.sol

```diff
@@ -398,42 +511,82 @@ contract Payroll is AragonApp { //, IForwarder { // makes coverage crash (remove
         uint128 employeeId = nextEmployee;
         employees[employeeId] = Employee({
             accountAddress: accountAddress,
-            denominationTokenSalary: initialYearlyDenominationSalary.toSecondDenominationToken(),
+            denominationTokenSalary: initialDenominationSalary,
+            accruedValue: 0,
             lastPayroll: startDate,
+            endDate: MAX_UINT64,
+            terminated: false,
             name: name
         });
         // Ids mapping
         employeeIds[accountAddress] = employeeId;
-        EmployeeAdded(
+        emit AddEmployee(
             employeeId,
             accountAddress,
-            initialYearlyDenominationSalary,
+            initialDenominationSalary,
             name,
             startDate
         );
         // update global variables
         nextEmployee++;
     }
 
+    function _addAccruedValue(uint128 employeeId, uint256 amount) internal {
+        employees[employeeId].accruedValue = employees[employeeId].accruedValue.add(amount);
+
+        emit AddEmployeeAccruedValue(employeeId, amount);
+    }
+
+    function _setPriceFeed(IFeed _feed) internal {
+        require(_feed != address(0));
+        feed = _feed;
+        emit SetPriceFeed(feed);
+    }
+
+    function _setRateExpiryTime(uint64 _time) internal {
+        require(_time > 0);
+        rateExpiryTime = _time;
+        emit SetRateExpiryTime(rateExpiryTime);
+    }
+
     /**
      * @dev Loop over tokens and send Payroll to employee
      * @param employeeId Id of employee receiving payroll
-     * @param time Time owed to employee (since last payroll)
      * @return True if something has been paid
      */
-    function _payTokens(uint128 employeeId, uint256 time) internal returns (bool somethingPaid) {
+    function _payTokens(uint128 employeeId) internal returns (bool somethingPaid) {
         Employee storage employee = employees[employeeId];
+
+        // get the min of current date and termination date
+        uint64 timestamp = getTimestamp64();
+        uint64 toDate;
+        if (employee.terminated && employee.endDate < timestamp) {
```

Comment:
I think this can just be `employee.endDate < timestamp`, since active employees have their `endDates` as `MAX_UINT64`.